### PR TITLE
Feature/separation of concerns

### PR DESCRIPTION
**Pull request by g-duff**
INTENT: Separation of concerns

* file paths, file formatting, numerical processing should be entirely separate from one another
* changing one concern (eg changing to a different microscope) should be possible without changing another concern (eg processing the image array)

### CODE REVIEW

File: src/analysis.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    selected_peak_locations = peak_locations[locations_sorted_by_prominence]
+
+    samplesize = len(row)
+    frequency_coordinates = np.fft.rfftfreq(samplesize, 1)
+    frequencies = frequency_coordinates[locations_sorted_by_prominence]
+
+    frequency_steps = [p / (micrometers_per_pixel * samplesize)
+                       for p in selected_peak_locations]
+    periods = [(1 / f) * 1E3 for f in frequency_steps]
+
```
Comment (g-duff) [inline src/analysis.py:65]: Is there some redundancy in lines 58 to 64? [59, 60] look quite similar to [62, 64].

**Review by jm1261 — APPROVED** [review]
Agreed changes, prefer lowercase variable names, camel case function names, underscored arguments, update about to be merged from dev

**Review by jm1261 — APPROVED** [review]
src directory not necessary but has been included, file name give as FileIO.py

**Review by jm1261 — APPROVED** [review]
originally had this as true, however doing a .json search in directory and a .json delete brought up the .txt output from SEM. This has been resolved by included a "results" directory, also much better practice to include a results directory to not alter directory/file trees

**Review by jm1261 — APPROVED** [review]
was roi for shorthand in further variable uses, but can see the appeal of having an easier variable name for ease of reading

**Review by jm1261 — APPROVED** [review]
agreed, altererd for readability to:

semparameters = SEMDict(lines=lines)
return semparameters

**Review by jm1261 — APPROVED** [review]
Nice tidy up here

**Pull request merged**